core/store/src/trie/mem/flexible_data/encoding.rs

```diff
@@ -0,0 +1,137 @@
+use crate::trie::mem::arena::{Arena, ArenaPtr, ArenaPtrMut, ArenaSliceMut};
+use borsh::{BorshDeserialize, BorshSerialize};
+use std::io::Write;
+
+use super::FlexibleDataHeader;
+
+/// Implementing this trait indicates that the type can be borsh serialized and
+/// the length of the serialization is a constant.
+///
+/// TODO: it would be nice to generate this, or test the correctness of the
+/// `SERIALIZED_SIZE` specification. For now, we rely on higher-level testing.
```

Comment:
lol wow - let me check this out :)

Reply:
Just curious if you had the time to check it out? :) (actually reading it again, it’ll probably be missing something like a `mod $typ {}` around `assert_borsh_fixed_size` in order to avoid naming conflicts, but this kind of issues should be reasonably easy to figure out)

Reply:
Also, a good thing about this macro is, that as it uses `bolero::check!`, it’d be run on our fuzzer infra as a fuzzer, so we’d have as good confidence as we theoretically could that the serialized size is correct :)